src/Analyzer.JsonRuleEngine/Expressions/AllOfExpression.cs

```diff
@@ -1,59 +1,30 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System;
-using System.Collections.Generic;
-using Microsoft.Azure.Templates.Analyzer.Types;
-
 namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Expressions
 {
     /// <summary>
     /// Represents an allOf expression in a JSON rule.
     /// </summary>
-    internal class AllOfExpression : Expression
+    internal class AllOfExpression : CompoundExpression
```

Comment:
In the docs we refer to allOf and anyOf as Structured Expressions. Should we change this to that same term here? @VeraBE and @JohnathonMohr

Reply:
I was thinking of something like StructuredListExpression, but I'm okay with either

Reply:
Yeah I was going to ask about the name, happy to change all references to StructuredListExpression.

Reply:
Ended up using StructuredExpression when doing the changes.